### DISCUSSION TITLE
add new measures to stats['_trades']

### DISCUSSION THREAD

**Discussion by Archkras** [kernc/backtesting.py#373] [Ideas]
Hello ,  thank you for a great package and for using clearly logic.
I want to add new measures for stats['_trades'] (variable take from your example) that show me a max possible profit on each trade (or highest price in lifetime of the trade) and max drawdown on each trade (or min value of price in lifetime of the trade ). What a best way to realize it, by your opinion?

**Comment by Benouare** [comment]
Hi,

I will do something like that to solve your probleme, 
```
def next(self):
        if not self.position:
            if self.data.index[-1].date() == date(2010, 5, 13):
                self.buy()
        else:
            # HERE
            if not hasattr(self.trades[-1], 'max_val') or self.trades[-1].max_val < self.position.pl_pct:
                            self.trades[-1].max_val = self.position.pl_pct
            # END HERE
            if self.data.index[-1].date() == date(2010, 5, 20):
                self.position.close()
```



```
bt = Backtest(data, Test, cash=10000, commission=0)
output = bt.run()
print(output._strategy.closed_trades[0].max_val)
```

**Reply by Archkras** [reply]
Thanks, it is very interesting. I think this is more for real-time decisions.

**Comment by kernc** [comment]
You could use [post-run trade data](https://kernc.github.io/backtesting.py/doc/examples/Quick%20Start%20User%20Guide.html#Trade-data) to compute those statistics after the fact, e.g.:

> highest price in lifetime of the trade

```py
bt = Backtest(data, ...)
stats = bt.run()
trades = states._trades

for trade in trades.itertuples():
    peak_high = data.High.iloc[trade.EntryBar:trade.ExitBar + 1]
    peak_high / trade.EntryPrice
    ...    
```

**Reply by Archkras** [reply]
Thanks for fast answer. This is exactly what  I need.
